### PR TITLE
Save CMS Block using repository

### PR DESCRIPTION
**Pull request by Jeroenll**
### Description
Update Save action of CMS block to be consistent with CMS Page.

### Fixed Issues (if relevant)
1. None

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/Cms/Controller/Adminhtml/Block/Save.php [hunk 1]
```diff
@@ -55,27 +73,33 @@ public function execute()
...
             /** @var \Magento\Cms\Model\Block $model */
-            $model = $this->_objectManager->create(\Magento\Cms\Model\Block::class)->load($id);
-            if (!$model->getId() && $id) {
-                $this->messageManager->addError(__('This block no longer exists.'));
-                return $resultRedirect->setPath('*/*/');
+            $model = $this->blockFactory->create();
+
+            $id = $this->getRequest()->getParam('block_id');
+            if ($id) {
+                $model->load($id);
```
Comment (dmanners) [inline app/code/Magento/Cms/Controller/Adminhtml/Block/Save.php]: Could we not also load the block from the repository?

Comment (dmanners) [reply app/code/Magento/Cms/Controller/Adminhtml/Block/Save.php]: Something like `$this->blockRepository->getById($id)`

**Review by dmanners — CHANGES_REQUESTED** [review]

File: app/code/Magento/Cms/Controller/Adminhtml/Block/Save.php (hunk 1 above)
Comment (Jeroenll) [reply app/code/Magento/Cms/Controller/Adminhtml/Block/Save.php]: Sure, give me 2 sec.

**Comment by Jeroenll** [comment]
@dmanners Updated 😄

**Comment by dmanners** [comment]
@JeroenVanLeusden you will also have to cover the unit tests updates as well. Since `app/code/Magento/Cms/Test/Unit/Controller/Adminhtml/Block/SaveTest.php` will now fail as it is written with the model in mind.

**Comment by Jeroenll** [comment]
Totally forgot that one, will update those tonight and ping you when ready.

**Comment by Jeroenll** [comment]
@dmanners Should be good to go now.

**Pull request merged**